nilearn/glm/second_level/second_level.py

```diff
@@ -42,7 +42,10 @@ def _check_second_level_input(second_level_input, design_matrix,
             raise ValueError('A second level model requires a list with at'
                              ' least two first level models or niimgs')
         # Check FirstLevelModel objects case
-        if flm_object and isinstance(second_level_input[0], FirstLevelModel):
+        if not flm_object and isinstance(second_level_input[0], FirstLevelModel):
+            raise ValueError(
+                "Contradictory arguments: flm_object is set to False yet second_level_input is a list of FirstLevelModel objects")
+        elif flm_object and isinstance(second_level_input[0], FirstLevelModel):
```

Comment:
You can simplify these conditions:

```python
if isinstance(second_level[0], FirstLevelModel):
    if flm_object:
        # DO STUFFS
    else:
        raise ValueError("bla bla")
```

Reply:
I simplified following as best I can your suggestion @NicolasGensollen. Tell me what you think !